### PR TITLE
Bug 222 - Unittests for login view

### PR DESCRIPTION
**Pull request by filipgorczynski**

### CODE REVIEW

File: volontulo/tests/views/test_auth.py [hunk 1]
```diff
@@ -125,3 +125,131 @@ def test__logged_out_authenticated_user(self):
...
+            response,
+            '/me',
+            302,
+            200,
+        )
+        self.assertEqual(len(response.redirect_chain), 1)
+        self.assertEqual(
+            response.redirect_chain[0],
+            ('http://testserver/me', 302),
+        )
```
Comment (magul) [inline volontulo/tests/views/test_auth.py:126]: You should also check session auth key.

File: volontulo/views/auth.py [hunk 2]
```diff
@@ -24,6 +24,10 @@
 
 def login(request):
     u"""Login view."""
+    if '_auth_user_id' in request.session:
+        return redirect('/me')
+
+    user_form = UserForm()
```
Comment (magul) [inline volontulo/views/auth.py:30]: is there any reason to set user_form variable?

File: volontulo/views/auth.py [hunk 3]
```diff
@@ -24,6 +24,10 @@
 
 def login(request):
     u"""Login view."""
+    if '_auth_user_id' in request.session:
```
Comment (magul) [inline volontulo/views/auth.py]: it should be done more pythonic way request.user.is_authorized()

File: volontulo/tests/views/test_auth.py [hunk 4]
```diff
@@ -125,3 +125,131 @@ def test__logged_out_authenticated_user(self):
             u"Użytkownik został wylogowany!"
         )
         self.assertNotIn('_auth_user_id', self.client.session)
+
```
Comment (magul) [inline volontulo/tests/views/test_auth.py]: Right now we have separate classes for every view - that should stay like that, so You need to create separate classes for TestLogin and TestLogout.

File: volontulo/tests/views/test_auth.py [hunk 5]
```diff
@@ -125,3 +125,131 @@ def test__logged_out_authenticated_user(self):
...
+    def test__get_login_by_anonymous(self):
+        u"""Get login form by anonymous user"""
```
Comment (magul) [inline volontulo/tests/views/test_auth.py:88]: full stop at end

File: volontulo/tests/views/test_auth.py [hunk 6]
```diff
@@ -125,3 +125,131 @@ def test__logged_out_authenticated_user(self):
...
+            response,
+            u'Email address:'
+        )
+        self.assertContains(
+            response,
+            u'Password:'
+        )
+
+    def test__get_login_by_authorized(self):
+        u"""Get login form by authorized user"""
```
Comment (magul) [inline volontulo/tests/views/test_auth.py:109]: ditto

File: volontulo/tests/views/test_auth.py [hunk 7]
```diff
@@ -125,3 +125,131 @@ def test__logged_out_authenticated_user(self):
...
+            200,
+        )
+        self.assertEqual(len(response.redirect_chain), 1)
+        self.assertEqual(
+            response.redirect_chain[0],
+            ('http://testserver/me', 302),
+        )
+
+    def test__post_login_by_anonymous_user(self):
+        u"""Post to login form by anonymous"""
```
Comment (magul) [inline volontulo/tests/views/test_auth.py:131]: ditto

File: volontulo/tests/views/test_auth.py [hunk 8]
```diff
@@ -125,3 +125,131 @@ def test__logged_out_authenticated_user(self):
...
+            200,
+        )
+        self.assertEqual(len(response.redirect_chain), 1)
+        self.assertEqual(
+            response.redirect_chain[0],
+            ('http://testserver/', 302),
+        )
+
+    def test__post_login_by_authorized_user(self):
+        u"""Post to login form by authorized"""
```
Comment (magul) [inline volontulo/tests/views/test_auth.py:199]: ditto

File: volontulo/views/auth.py (hunk 2 above)
Comment (filipgorczynski) [inline volontulo/views/auth.py:30]: It's because related model class. Without this rendered form has no labels for email and password.

**Pull request merged**